media-production/use-cases-and-gap-analysis-frame-accurate-sync.md

```diff
@@ -0,0 +1,43 @@
+# Use Cases and Gap Analysis: Frame Accuracy for Media Production
+
+This document collects use cases and requirements for new or changed Web APIs related to frame accuracy to improve support for media production.
+
+## Functional Gaps
+
+### The media timeline is not precise enough to identify individual frames
+
+[HTML] does not expose any precise mechanism to assess the time, from a user's wall clock perspective, at which a particular media frame is going to be rendered. A web application may only infer this information by looking at the [media element]'s [currentTime] property to infer the frame being rendered and the time at which the user will see the next frame. This has several limitations:
+
+* [currentTime] is represented as a double value, which does not allow it to identify individual frames due to rounding errors. This is a [known issue][html#109].
+* [currentTime] is updated at a user-agent defined rate (typically the rate at which [time marches on] runs), and is kept stable while scripts are running. When a web application reads [currentTime], it cannot tell when this property was last updated, and thus cannot reliably assess whether this property still represents the frame currently being rendered.
+
+In addition, [currentTime] only accepts a time value, and neither has the abiliy, nor is there an alternative API, to set its value to a frame number or [SMPTE Time Code].
+
+### Seeking to next/previous frame
+
+The [media element] does not provide a mechanism to seek by individual frames. This can be worked around by using the media's frame rate and the [media element]'s [currentTime] property to seek by a frame duration from the reported [currentTime] value, but the [currentTime] property does not guarantee frame level precision. In addition, the frame rate of the media may vary over time, may be rounded internally by the browser, and is not exposed.
```

Comment:
If you asked "advance by 1 frame" across a frame rate change, would it be acceptable for the presentation to advance by 1 frame but the new `currentTime` to reflect the time according to the combined frame rates of all the frames including this new frame? I.e. mathematically integrate frame rate over the number of frames.